shell/common/rasterizer.cc

```diff
@@ -604,6 +604,9 @@ RasterStatus Rasterizer::DrawToSurfaceUnsafe(
     }
 
     SurfaceFrame::SubmitInfo submit_info;
+    // TODO (kaushikiska): this can be in the past and might need to get snapped
+    // to future as this frame could have been resubmitted.
+    submit_info.presentation_time = frame_timings_recorder.GetVsyncTargetTime();
```

Comment:
Bug?

Should we avoid setting this if it's a resubmit?

Reply:
filed an issue. for now i'll set this only if it is `> now()`, I think it should be the case for majority of frames. Once the issue is addressed we can snap this to the next vsync target time,.